pom.xml

```diff
@@ -187,6 +187,32 @@
                 <extensions>true</extensions>
             </plugin>
 
+            <plugin>
+                <groupId>org.moditect</groupId>
+                <artifactId>moditect-maven-plugin</artifactId>
+                <version>1.2.2.Final</version>
+                <executions>
+                    <execution>
+                        <id>add-module-infos</id>
+                        <goals>
+                            <goal>add-module-info</goal>
+                        </goals>
+                        <phase>package</phase>
+                        <configuration>
+                            <jvmVersion>9</jvmVersion>
+                            <module>
+                                <moduleInfo>
+                                    <name>com.pivovarit.function</name>
+                                    <exports>
+                                        com.pivovarit.function;
+                                    </exports>
+                                </moduleInfo>
```

Comment:
Instead of `<moduleInfo>`, we should use:
```xml
<moduleInfoSource>
module com.pivovarit.function {
    exports com.pivovarit.function;
}
</moduleInfoSource>
```

then it will also easily build on JDK8

Reply:
You should also remove ```<Automatic-Module-Name>com.pivovarit.function</Automatic-Module-Name>``` from `maven-jar-plugin` configuration